File: app/src/main/java/mitya/yahnc/ui/NewStoriesFragment.java
package mitya.yahnc.ui;

import mitya.yahnc.network.StoryService;
import rx.Observable;
import rx.schedulers.Schedulers;

/**
 * Created by Mitya on 25.08.2016.
 */
public class NewStoriesFragment extends RemoteStoryFragment {
    public NewStoriesFragment() {
        super(StoryService.getInstance().getNewStories().
                flatMap(stories -> Observable.from(stories).subscribeOn(Schedulers.io())));
    }
}

File: app/src/main/java/mitya/yahnc/ui/NewStoriesFragment.kt
package mitya.yahnc.ui

import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import mitya.yahnc.network.HnService

/**
 * Created by Mitya on 25.08.2016.
 */
class NewStoriesFragment : RemoteStoryFragment() {
    override fun getCurrentStories(): Observable<Int> {
        return HnService.getNewStories().flatMap { stories -> Observable.fromArray(*stories).subscribeOn(Schedulers.io()) }
    }
}
